types/wlr_output.c

```diff
@@ -182,16 +186,44 @@ void wlr_output_transform(struct wlr_output *output,
 
 bool wlr_output_set_cursor(struct wlr_output *output,
 		const uint8_t *buf, int32_t stride, uint32_t width, uint32_t height) {
-	if (!output->impl || !output->impl->set_cursor) {
-		return false;
+	if (output->impl->set_cursor && output->impl->set_cursor(output->state, buf,
+			stride, width, height)) {
+		output->cursor.is_sw = false;
+		return true;
+	}
+
+	wlr_log(L_INFO, "Falling back to software cursor");
+
+	output->cursor.is_sw = true;
+	output->cursor.width = width;
+	output->cursor.height = height;
+
+	if (!output->cursor.renderer) {
+		output->cursor.renderer = wlr_gles2_renderer_init();
```

Comment:
Instead of this, let's add `wlr_output_attach_renderer` and let the user provide one. Software cursor would just fail with a warning if there is no renderer attached.